File: backend/Models/Match.cs
using System.ComponentModel.DataAnnotations;

namespace FairPlay.Api.Models;

public class Match
{
    public Guid Id { get; set; }
    
    [Required]
    public Guid LeagueId { get; set; }
    
    public DateTime Date { get; set; }
    
    [MaxLength(200)]
    public string? Location { get; set; }
    
    [Required]
    [MaxLength(50)]
    public string FormatType { get; set; } = "8v8";
    
    public bool IsCompleted { get; set; }
    
    // Navigation properties
    public League? League { get; set; }
    public List<MatchAssignment> MatchAssignments { get; set; } = new();
    public List<RawRating> RawRatings { get; set; } = new();
}

File: frontend/types/match.ts
import type { Player } from './player'

/** Match assignment linking a player to a team */
export interface MatchAssignment {
    playerId: string
    teamNumber: number
    player?: Player
}

/** Match as returned by GET /api/matches/:id */
export interface Match {
    id: string
    leagueId: string
    date: string
    location: string | null
    formatType: string
    isCompleted: boolean
    allowRatings: boolean | null
    matchAssignments: MatchAssignment[]
}

/** Dashboard data as returned by GET /api/matches/dashboard */
export interface DashboardData {
    lastCompletedMatchDate: string | null
    nextActiveMatchDate: string | null
    pendingRatingMatchDate?: string | null
    recentPerformance: { date: string; value: number }[]
    needsLeague?: boolean
}

/** Raw rating as returned by GET /api/matches/:id/my-ratings */
export interface RawRating {
    subjectId: string
    value: number
}
